pkg/resource/deploy/step_generator.go

```diff
@@ -379,21 +379,13 @@ func (sg *stepGenerator) GenerateSteps(
 		return []Step{NewSameStep(sg.plan, event, old, new)}, nil
 	}
 
-	if updateTargetsOpt != nil && !updateTargetsOpt[urn] {
-		d := diag.GetResourceIsBeingCreatedButWasNotSpecifiedInTargetList(urn)
-
-		if sg.plan.preview {
-			// During preview we only warn here but let the planning proceed.  This allows the user
-			// to hear about *all* the potential resources they'd need to add a -target arg for.
-			// This prevents the annoying scenario where you get notified about a problem, fix it,
-			// rerun and then get notified about the very next problem.
-			sg.plan.Diag().Warningf(d, urn)
-		} else {
-			// Targets were specified, but didn't include this resource to create.  Give a particular
-			// error in that case and stop immediately.
-			sg.plan.Diag().Errorf(d, urn)
-			return nil, result.Bail()
-		}
+	if updateTargetsOpt != nil && !updateTargetsOpt[urn] && !sg.opts.TargetDependents {
```

Comment:
Hm. This part of the behavior doesn't really have anything to do with dependents... Maybe the new name isn't so great?

Also, I think we should still be warning during preview and failing during the update.

Reply:
can we take the discussion about desired behavior to: https://github.com/pulumi/pulumi/issues/3369

I did this PR based on the conclusions there so far.  If we want to revise them, we can.  

I'm personally trying to drive this from the perspective of the user.  Effectively, i think it makes sense from teh user perspective to have these separate buckets:

1. resources i explicitly mentioned that should be touched.
2. resources that end up needing to be touched that i didn't mention.

The latter group can come about through several operations and it would be nice to have a *consistent* way to deal with them.  One of those consistent ways is:

a. just force the user to specify the resource explicitly in the '1' group, and otherwise fail the operation.

However, this opens up the question of user experience around forcing the user to have to go and manually specify a bunch of resources manually.  Should there be a way to make that easier?  And can that be done in a *consistent* fashion so the user doesn't have to learn a bunch of different ways to do things per-command.  In other words, they already just think of the "--target" as a uniform way to specify targets for a few ops.  So can there be a `--something` that is the uniform way of saying "ok ok, i get it, there are some other resources as well, go ahead and make those happen as well".?

Reply:
I think it's absolutely ok to have a flag that says "do all these things". I also think that if that flag is not specified, we should continue to warn on all resources and allow planning to proceed.

Reply:
> we should continue to warn on all resources and allow planning to proceed.

the strange thing about this though is that we then warn... but if the user says they want to go ahead, then it just fails.  

One way around this perhaps is if we *error*, but still proceed through the preview, but then finally error-out at the end.  That way the user hears about all the issues, but this is still considered an error they need to take an affirmative step to resolve (i.e. either by adding more `--target`s, or by adding the `--do-it-damnit` flag).

This would still satisfy all the goals i have above.  i.e.

1. not mentioning a resource is a problem that the user must acknowledge and provide guidance on
2. providing guidance can be done in a non-onerous way.

It also improves on things by ensuring that users can hear about the complete set of issues up-front during preview without having to rerun the preview multiple times.

Would that be amenable to you?

Reply:
Yes, that seems reasonable.

Reply:
Ok.  have taken this approach.